Panic on startup  runtime error: slice bounds out of range hh/processor.go:200

Got this panic on startup running version: 8c715f7
I have the opentsdb plugin and am running 3 servers in cluster mode.

One node crashed and everytime on start up now it panics with this error. Below is the full startup log.
```
[metastore] 2015/07/22 20:21:31 [INFO] raft: Node at 10.240.50.65:8088 [Follower] entering Follower state
[metastore] 2015/07/22 20:21:31 read local node id: 1
[metastore] 2015/07/22 20:21:32 [WARN] raft: Heartbeat timeout reached, starting election
[metastore] 2015/07/22 20:21:32 [INFO] raft: Node at 10.240.50.65:8088 [Candidate] entering Candidate state
[metastore] 2015/07/22 20:21:32 [DEBUG] raft: Votes needed: 2
[metastore] 2015/07/22 20:21:32 [DEBUG] raft: Vote granted. Tally: 1
[httpd] 2015/07/22 20:21:32 listening on HTTP: [::]:8086
panic: runtime error: slice bounds out of range

goroutine 35 [running]:
github.com/influxdb/influxdb/services/hh.(*Processor).unmarshalWrite(0xc208044660, 0xe93788, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.92T9jQoUcC/src/github.com/influxdb/influxdb/services/hh/processor.go:200 +0x160
github.com/influxdb/influxdb/services/hh.func·002(0x3, 0xc2080446c0)
        /tmp/tmp.92T9jQoUcC/src/github.com/influxdb/influxdb/services/hh/processor.go:146 +0x20b
created by github.com/influxdb/influxdb/services/hh.(*Processor).Process
        /tmp/tmp.92T9jQoUcC/src/github.com/influxdb/influxdb/services/hh/processor.go:177 +0x1f7

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@
 - [#3579](https://github.com/influxdb/influxdb/issues/3579): Revert breaking change to `client.NewClient` function
 - [#3580](https://github.com/influxdb/influxdb/issues/3580): Do not allow wildcards with fields in select statements
 - [#3530](https://github.com/influxdb/influxdb/pull/3530): Aliasing a column no longer works
+- [#3436](https://github.com/influxdb/influxdb/issues/3436): Fix panic in hinted handoff queue processor
 
 ## v0.9.2 [2015-07-24]
 

--- a/services/hh/processor.go
+++ b/services/hh/processor.go
@@ -145,10 +145,10 @@ func (p *Processor) Process() error {
 				// unmarshal the byte slice back to shard ID and points
 				shardID, points, err := p.unmarshalWrite(buf)
 				if err != nil {
-					// TODO: If we ever get and error here, we should probably drop the
-					// the write and let anti-entropy resolve it.  This would be an urecoverable
-					// error and could block the queue indefinitely.
-					res <- err
+					p.Logger.Printf("unmarshal write failed: %v", err)
+					if err := q.Advance(); err != nil {
+						res <- err
+					}
 					return
 				}
 
@@ -197,6 +197,9 @@ func (p *Processor) marshalWrite(shardID uint64, points []tsdb.Point) []byte {
 }
 
 func (p *Processor) unmarshalWrite(b []byte) (uint64, []tsdb.Point, error) {
+	if len(b) < 8 {
+		return 0, nil, fmt.Errorf("too short: len = %d", len(b))
+	}
 	ownerID := binary.BigEndian.Uint64(b[:8])
 	points, err := tsdb.ParsePoints(b[8:])
 	return ownerID, points, err

